src/main/java/org/bookstore/mapper/BookMapper.java

```diff
@@ -16,7 +17,9 @@
 public interface BookMapper {
     BookDto toDto(Book book);
 
-    Book toBook(CreateBookRequestDto bookRequestDto);
+    Book toEntity(CreateBookRequestDto bookDto);
+
+    BookDtoWithoutCategoryIds toDtoWithoutCategories(Book book);
```

Comment:
you should add `default void setCategoryIds(@MappingTarget BookDto bookDto, Book book)` (from your task)